pkg/storegateway/threadpool/threadpool.go

```diff
@@ -71,38 +69,33 @@ func NewThreadPool(num int, reg prometheus.Registerer) *Threadpool {
 		tp.pool <- t
 	}
```

Comment:
I think it would be cleaner to move this to starting function since this is a service now, WDYT?

(Not a blocking comment)

Reply:
Good idea. I'll see if I can make that work :+1:

Reply:
Fixed in 88bd9e77679739162f6bbec4ac28983304ddbf78